index.js

```diff
@@ -12,11 +12,15 @@ const { log } = console;
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 
-app.use('/api/v1', routes);
+app.get('/', (req, res) => {
+  res.status(200).json({
+    message: 'Hello Books Deferral'
+  });
+});
 
-app.use('*', (req, res) => {
+app.all('*', (req, res) => {
```

Comment:
The catch-all here is really cool, is there any particular difference between `app.use` and `app.all`?

Reply:
@tolulope-od  Thanks, once again, for the feedback. I feel `app.all` is more suitable for route handling while `app.use` is more suitable for middlewares. Express route handlers are **technically** middlewares though. This is strictly my opinion, so if you have better suggestions, I look forward to learning from you.

You might as well read [this discussion on Stack Overflow](https://stackoverflow.com/questions/14125997/difference-between-app-all-and-app-use)

Reply:
Thanks for explaining this @abdulfatai360 I have seen them used differently in separate apps and I just wanted some clarity on the difference